driver_test.go

```diff
@@ -181,88 +193,96 @@ func TestQueryContext(t *testing.T) {
 		VALUES ("a1", "b1", "c1"), ("a2", "b2", "c2") , ("a3", "b3", "c3") `})
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
+	}
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
 	}
+	defer db.Close()
 
 	type testQueryContextRow struct {
 		A, B, C string
 	}
 
 	tests := []struct {
+		name           string
 		input          string
 		want           []testQueryContextRow
-		wantErrorQuery string
-		wantErrorScan  string
+		wantErrorQuery bool
+		wantErrorScan  bool
 	}{
-		// empty query
-		{input: "", want: []testQueryContextRow{}},
-		// syntax error
-		{input: "SELECT SELECT * FROM TestQueryContext", want: []testQueryContextRow{}},
-		// retur nothing
-		{input: "SELECT SELECT * FROM TestQueryContext", want: []testQueryContextRow{}},
-		// return one tuple
-		{input: "SELECT * FROM TestQueryContext WHERE A = \"a1\"",
+		{name: "empty query",
+			input: "", want: []testQueryContextRow{}},
+		{name: "suntax error",
+			input: "SELECT SELECT * FROM TestQueryContext", want: []testQueryContextRow{}},
+		{name: "return nothing",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"hihihi\"", want: []testQueryContextRow{}},
+		{name: "select one tuple",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"a1\"",
 			want: []testQueryContextRow{
 				{A: "a1", B: "b1", C: "c1"},
 			}},
-		// return subset of tuples
-		{input: "SELECT * FROM TestQueryContext WHERE A = \"a1\" OR A = \"a2\"",
+		{name: "select subset of tuples",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"a1\" OR A = \"a2\"",
 			want: []testQueryContextRow{
 				{A: "a1", B: "b1", C: "c1"},
 				{A: "a2", B: "b2", C: "c2"},
 			}},
-		// subet of tuples with !=
-		{input: "SELECT * FROM TestQueryContext WHERE A != \"a3\"",
+		{name: "select subset of tuples with !=",
+			input: "SELECT * FROM TestQueryContext WHERE A != \"a3\"",
 			want: []testQueryContextRow{
 				{A: "a1", B: "b1", C: "c1"},
 				{A: "a2", B: "b2", C: "c2"},
 			}},
-		// return entire table
-		{input: "SELECT * FROM TestQueryContext ORDER BY A",
+		{name: "select entire table",
+			input:         "SELECT * FROM TestQueryContext ORDER BY A",
 			want: []testQueryContextRow{
 				{A: "a1", B: "b1", C: "c1"},
 				{A: "a2", B: "b2", C: "c2"},
 				{A: "a3", B: "b3", C: "c3"},
 			}},
-		// query non existant table
-		{input: "SELECT * FROM TestQueryContexta", want: []testQueryContextRow{}},
-	}
-
-	// Open db.
-	ctx := context.Background()
-	db, err := sql.Open("spanner", dsn)
-	if err != nil {
-		debug.PrintStack()
-		log.Fatal(err)
+		{name: "query non existant table",
+			input: "SELECT * FROM TestQueryContexta", want: []testQueryContextRow{}},
 	}
-	defer db.Close()
 
 	// Run tests
 	for _, tc := range tests {
 
 		rows, err := db.QueryContext(ctx, tc.input)
-		if !ErrorContainsStr(err, tc.wantErrorQuery) {
-			t.Errorf("Unexpected error %v", err)
+		if !ErrorExpected(err, tc.wantErrorQuery) {
```

Comment:
Rather than creating a function and using nested if statements, use two separate if statements. One for when you are expecting an error, and one for when you aren't. It's much more readable and it's clear what the conditions are e.g. `if err != nil && !tc.wantErrorQuery` clearly checks for unexpected errors

Reply:
I have no idea what I was going for with this & choose to blame the heat wave. I also added the rows.close error which doesn't throw anything in any of these cases